Feature/dccprtl 330 new pql endpoints should be able to return data cleaned of elasticsearch related information

KevinFHartmann opened the pull request.

KevinFHartmann committed:
Enhancement: Refactored createResponseMap() in ElasticSearchRepositoryUtils and it's internal helper functions to de-couple a Query object from it's include
list.

Next Steps: Code cleanup, and then implementation of a PQL query that uses the
new createResponseMap with an appropriately generated include list.

KevinFHartmann committed:
Code Cleanup: Fixed spelling of "flatten" (was "flattern").
Code Cleanup: Improved wording of comment.

KevinFHartmann committed:
Enhancement: Made the PQL service for Donor return Donor objects.

KevinFHartmann committed:
Enhancement: Added a static constructor for Pagination of pql queries.
Enhancement: All of the entities now return the appropriate entity type.
Issues Remaining: (1) GeneService still needs to add code for PQL handling of filters.
(2) Code needs refactoring & cleanup.

KevinFHartmann committed:
Code Cleanup: Factored out the part of a Pagination object that is known when a request is made into a separate PaginationRequest object, so that it can be passed in and filled in later.
Code Cleanup: Made all of the entities use named constants instead of bare strings, refactored and generally cleaned up and simplified code.
Enhancement: Wrote the code to match project ids for PQL instead of Query objects in the Gene service

KevinFHartmann committed:
Bugfix: For Occurrences, use findAll in OccuranceRepository. In OccuranceRepository, make PQL and Query objects use exactly the same search query.
Code Cleanup: Changed name from FindAllCentric to FindAll for the PQL query to match the Query object query.
Code Cleanup: Made buildFindAllRequest() have a PQL counterpart, for consistency.

andricDu commented on dcc-portal-server/src/main/java/org/icgc/dcc/portal/server/model/Pagination.java:
```diff
@@ -22,50 +22,43 @@
-  private Pagination(Integer count, Long total, Query query) {
-    this(count, total, query.getSize(), query.getFrom(), query.getSort(), query.getOrder().toString());
-  }
+//  public static Pagination of(int count, long total, int size, int from, @NonNull String sort, @NonNull String order) {
```
If this is no longer needed, delete.

andricDu commented on dcc-portal-server/src/main/java/org/icgc/dcc/portal/server/model/PaginationRequest.java:
```diff
@@ -0,0 +1,61 @@
+
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class PaginationRequest {
+		@ApiModelProperty(value = "Number of records to return in this batch", required = true)
```
tabs are being used instead of spaces.

andricDu commented on dcc-portal-server/src/main/java/org/icgc/dcc/portal/server/resource/entity/MutationResource.java:
```diff
@@ -98,14 +100,15 @@ public Mutations findAll(
   @Path("/pql")
-  @ApiOperation(value = "String")
-  public String findPQL(
+
```
you have an extra newline here.

andricDu commented on dcc-portal-server/src/main/java/org/icgc/dcc/portal/server/resource/entity/OccurrenceResource.java:
```diff
@@ -71,14 +72,14 @@ public Occurrences findAll(
   @Path("/pql")
-  @ApiOperation(value = "String")
-  public String findPQL(
+  @ApiOperation(value = RETURNS_LIST + OCCURRENCE + S, response = Projects.class)
```
Is `Projects.class` correct?

dcc-jenkins commented:
Deployed to https://dev.dcc.icgc.org/portals/433 for testing

KevinFHartmann committed:
Bugfix: Changes as per PR

dcc-jenkins commented:
Deployed to https://dev.dcc.icgc.org/portals/434 for testing

andricDu commented:
Keep PR open for testing on dev.

KevinFHartmann committed:
Enhancement: Added endpoints for PQL downloads

andricDu commented:
@rtisma you mind fixing the compilation error on the importer?

andricDu commented:
Looks like this may be causing an issue in the UI pagination. When in advanced search I click next page and it appears the `from` value is set to `null`

localhost:8080/search/m?donors={"size":50,"from":null}

andricDu commented:
It looks like on the initial page load the pagination response is different.

Old response when loading the donor tab for the first time:
![screen shot 2017-11-21 at 4 57 18 pm](https://user-images.githubusercontent.com/6350043/33098967-2ce072ea-cedd-11e7-9723-d17257cb7a7b.png)

New response:
![screen shot 2017-11-21 at 4 57 30 pm](https://user-images.githubusercontent.com/6350043/33098975-3238b252-cedd-11e7-8f0e-ecc3a66d969d.png)

dcc-jenkins commented:
Deployed to https://dev.dcc.icgc.org/portals/435 for testing

KevinFHartmann committed:
Bugfix: Gave the Pagination model class back the getters that I accidentally refactored out,and made it serialize to JSON properly.
Bugfix: Queries that have facetsOnly can still submit sort information; PQL can't represent such the sort information, but we still want to return it back to
the web portal, so we set the pagination info from the original query before we return our results.

dcc-jenkins commented:
Deployed to https://dev.dcc.icgc.org/portals/441 for testing
